World/WeatherManager.gd

```diff
@@ -40,25 +51,38 @@ var wind_direction = 0 :
 		return wind_direction
 	set(new_wind_direction):
 		wind_direction = new_wind_direction
-		emit_signal("wind_direction_changed", wind_direction)
+		wind_direction_changed.emit(wind_direction)
 
 var rain_enabled := false :
 	get:
 		return rain_enabled
 	set(enabled):
 		rain_enabled = enabled 
-		emit_signal("rain_enabled_changed", enabled)
+		rain_enabled_changed.emit(enabled)
 
-var rain_density := 100.0 :
+var rain_density := 1 :
 	get:
 		return rain_density
 	set(new_rain_density):
 		rain_density = new_rain_density 
-		emit_signal("rain_density_changed", rain_density)
+		rain_density_changed.emit(rain_density)
 
-var rain_drop_size := Vector2(0.5, 0.25) :
+var rain_drop_size := 0.1 :
 	get:
 		return rain_drop_size
 	set(new_rain_drop_size):
 		rain_drop_size = new_rain_drop_size
-		emit_signal("rain_drop_size_changed", new_rain_drop_size)
+		rain_drop_size_changed.emit(new_rain_drop_size)
+
+var lightning_enabled := false :
+	get:
+		return lightning_enabled
+	set(enabled):
+		lightning_enabled = enabled
+		lightning_enabled_changed.emit(lightning_enabled)
+
+
+var lightning_rotation := 0 :
```

Comment:
Not really intuitive what this means (rotation relative to what? what's being rotated? deg or rad?)